### ISSUE TITLE
No preference with the name 'tempfiles.recovery.checkforrecoveries' was found. 

### ISSUE THREAD

**Issue by Daivesd** [MPh-py/MPh#39]
Hi there! 
First of all thanks for the great work you've done ! 
I was trying to setup a simulation using your package, but I was not able to start any client... 
I'm using Manjaro Linux as operating system, and the MATLAB livelink interface seems to work fairly well. 
Here's the error report: 

```none
Exception                                 Traceback (most recent call last)
/usr/lib/python3.9/site-packages/_jpype.cpython-39-x86_64-linux-gnu.so in java.lang.Thread.run()
/usr/lib/python3.9/site-packages/_jpype.cpython-39-x86_64-linux-gnu.so in java.util.concurrent.ThreadPoolExecutor$Worker.run()
/usr/lib/python3.9/site-packages/_jpype.cpython-39-x86_64-linux-gnu.so in java.util.concurrent.ThreadPoolExecutor.runWorker()
/usr/lib/python3.9/site-packages/_jpype.cpython-39-x86_64-linux-gnu.so in java.util.concurrent.FutureTask.run()
/usr/lib/python3.9/site-packages/_jpype.cpython-39-x86_64-linux-gnu.so in com.comsol.bridge.command.j$3.call()
/usr/lib/python3.9/site-packages/_jpype.cpython-39-x86_64-linux-gnu.so in com.comsol.bridge.command.j$3.a()
/usr/lib/python3.9/site-packages/_jpype.cpython-39-x86_64-linux-gnu.so in com.comsol.bridge.command.j.a()
/usr/lib/python3.9/site-packages/_jpype.cpython-39-x86_64-linux-gnu.so in com.comsol.bridge.command.j.d()
/usr/lib/python3.9/site-packages/_jpype.cpython-39-x86_64-linux-gnu.so in com.comsol.bridge.command.i.execute()
/usr/lib/python3.9/site-packages/_jpype.cpython-39-x86_64-linux-gnu.so in com.comsol.bridge.command.c.run()
/usr/lib/python3.9/site-packages/_jpype.cpython-39-x86_64-linux-gnu.so in com.comsol.bridge.command.b.a()
/usr/lib/python3.9/site-packages/_jpype.cpython-39-x86_64-linux-gnu.so in java.lang.reflect.Method.invoke()
/usr/lib/python3.9/site-packages/_jpype.cpython-39-x86_64-linux-gnu.so in sun.reflect.DelegatingMethodAccessorImpl.invoke()
/usr/lib/python3.9/site-packages/_jpype.cpython-39-x86_64-linux-gnu.so in sun.reflect.NativeMethodAccessorImpl.invoke()
/usr/lib/python3.9/site-packages/_jpype.cpython-39-x86_64-linux-gnu.so in sun.reflect.NativeMethodAccessorImpl.invoke0()
/usr/lib/python3.9/site-packages/_jpype.cpython-39-x86_64-linux-gnu.so in com.comsol.model.util.ModelUtil.setPreference()
/usr/lib/python3.9/site-packages/_jpype.cpython-39-x86_64-linux-gnu.so in com.comsol.model.util.ServerModelUtil.setPreference()
/usr/lib/python3.9/site-packages/_jpype.cpython-39-x86_64-linux-gnu.so in java.security.AccessController.doPrivileged()
/usr/lib/python3.9/site-packages/_jpype.cpython-39-x86_64-linux-gnu.so in com.comsol.model.util.ServerModelUtil$21.run()
/usr/lib/python3.9/site-packages/_jpype.cpython-39-x86_64-linux-gnu.so in com.comsol.nativeutil.preferences.Preferences.setPreference()

Exception: Java Exception

The above exception was the direct cause of the following exception:

com.comsol.util.exceptions.FlException    Traceback (most recent call last)
<ipython-input-3-9a0936dc3385> in <module>
----> 1 client = mph.start(cores=1)

/usr/lib/python3.9/site-packages/mph/session.py in start(cores, version, port)
    104     elif session == 'client-server':
    105         server = Server(cores=cores, version=version, port=port)
--> 106         client = Client(cores=cores, version=version, port=server.port)
    107     else:
    108         error = f'Invalid session type "{session}".'

/usr/lib/python3.9/site-packages/mph/client.py in __init__(self, cores, version, port, host)
    138         java.setPreference('tempfiles.saving.warnifoverwriteolder', 'off')
    139         java.setPreference('tempfiles.recovery.autosave', 'off')
--> 140         java.setPreference('tempfiles.recovery.checkforrecoveries', 'off')
    141         java.setPreference('tempfiles.saving.optimize', 'filesize')
    142 

com.comsol.util.exceptions.FlException: Exception:
        com.comsol.util.exceptions.FlException: No preference with the name X was found#tempfiles.recovery.checkforrecoveries
Messages:
        No preference with the name 'tempfiles.recovery.checkforrecoveries' was found.
```

I really have no idea of what this exception is; I've read the 'Limitations' section in the documentation but this error seems different to the one reported there... 
Any idea of what's happening ? Am I doing something wrong ?
Thanks!

**Comment by John-Hennig** [comment]
Hi, thanks a lot for reporting this. No, you didn't do anything wrong. This one's entirely on me.

See, in earlier versions, this error was anticipated and handled internally. The handling of that error was introduced in an old pull request, not by me, but there was no comment in the code why it was even needed. So I removed it: 1ae89ef069b2026826d95dda93e39528ea5132cd (line 127). That was shortly before releasing version 1.0. I figured, if it breaks, someone will eventually report it. 🤷 And here you are! 😉 

I did however test the change on Ubuntu (18.04 LTS and 20.04), which is [officially supported by Comsol](https://www.comsol.com/support/knowledgebase/1286), whereas Manjaro is not. I actually like Manjaro better, I just never even tried if Comsol can be installed there *because* it's not supported. Maybe errors such as this one have something to do with that. Though I have no idea why certain Comsol preferences would be available on one Linux flavor, but not on another.

Anyway, this is a very easy fix. I would however like to include yet another bug fix in the next release (1.0.4), but there we are waiting for a new release of an external dependency (JPype). So, I'd rather wait a little longer.

Your immediate options to fix this are:
* Use MPh 0.9.1 (if you don't need the newer features).
* Hack your local copy of the installed library code by adding lines 127 to 130 (in the diff) back in `client.py`.

**Comment by Daivesd** [comment]
Ok I see thank you very much! I'll roll back to 0.9.1 since I'm just testing stuff right now :smile: 

I guess they stick to Debian-based distros since the rolling release is much harder to mantain as a stable package. Indeed, everything works just as fine as in any other linux distribution, since the installation does not really rely on specific os commands, afaik ^^ turns out this is not completely true since on my system something broke ahah

Anyway, let me thank you again for this project, I find it amazing ! It's not much help but I'll keep testing it on linux here :)

**Comment by John-Hennig** [comment]
Any help is welcome and appreciated. And I'm glad you see the potential of this library. It was a huge productivity boost for me when I realized I can automate my simulations from Python (thanks to JPype, really). Matlab is good for a lot of scientific computing, but quite clunky when it comes to automation. And Java is just way too verbose. It takes like ten lines of Java code for every line of Python. Plus the compilation steps, it's just too much. Scientists and engineers should focus on the science and the engineering, and spend as little time as possible on figuring out the software tools.

**Comment by max3-2** [comment]
Let me tag in here real quick: Im getting this on windows with `python 3.7.5` and a classkit installation. We should have a solution or at least a try except asap since there seems to be more than one situation where this can occur...

**Comment by John-Hennig** [comment]
Fixed in version 1.0.4, released today.
